crates/markdown_preview/src/markdown_parser.rs

```diff
@@ -300,10 +303,39 @@ impl<'a> MarkdownParser<'a> {
                             }
                         }
                         if new_highlight {
-                            highlights
-                                .push((last_run_len..text.len(), MarkdownHighlight::Style(style)));
+                            highlights.push((
+                                last_run_len..text.len(),
+                                MarkdownHighlight::Style(style.clone()),
+                            ));
                         }
                     }
+                    if let Some(image) = image.clone() {
+                        let is_valid_image = match image.clone() {
+                            Image::Path {
+                                display_path,
+                                path: _,
+                            } => display_path.is_file(),
+                            Image::Web { url } => match isahc::get(url) {
```

Comment:
I don't think we want to use this isahc::get API here. Rather, GPUI already has an `ImageSource` type that you can pass directly to `img()`, that will make the HTTP request or read the file for you. Maybe just parse the image tag into this type and use it?

Reply:
yeah I will do that then and commit the changes

Reply:
> I don't think we want to use this isahc::get API here. Rather, GPUI already has an `ImageSource` type that you can pass directly to `img()`, that will make the HTTP request or read the file for you. Maybe just parse the image tag into this type and use it?
 yeah the reason why I am using isahc get is to see whether the image is valid as per markdown spec if the image fails to get loaded we have to render the all source as of now GPUI Image doesn't return an error or any kind of alt text if the image fails to get loaded that's why I am adding a check here for now. any suggestions are welcome 😄